src/entries/popup/pages/home/Activity/ActivityDetails.tsx

```diff
@@ -153,29 +154,41 @@ const InfoValueSkeleton = () => <Skeleton width="50px" height="12px" />;
 function FeeData({ transaction: tx }: { transaction: RainbowTransaction }) {
   const { native, feeType } = tx;
 
-  const maxPriorityFeePerGas =
-    tx.maxPriorityFeePerGas && formatUnits(tx.maxPriorityFeePerGas, 'gwei');
-  const maxFeePerGas = tx.maxFeePerGas && formatUnits(tx.maxFeePerGas, 'gwei');
-  const baseFee = tx.baseFee && formatUnits(tx.baseFee, 'gwei');
+  // if baseFee is undefined (like in pending txs or custom networks the api wont have data about it)
+  // so we try to calculate with the data we may have locally
+  const baseFee =
+    tx.baseFee ||
+    (tx.maxFeePerGas &&
+      tx.maxPriorityFeePerGas &&
+      BigNumber.from(tx.maxFeePerGas).sub(tx.maxPriorityFeePerGas).toString());
 
-  const gasPrice = tx.gasPrice && formatUnits(tx.gasPrice, 'gwei');
+  let fee;
+  if (native !== undefined && native.fee !== undefined) {
+    // if the fee is less than $0.01, the provider returns 0 so we display it as <$0.01
+    const feeInNative = +native.fee <= 0.01 ? 0.01 : native.fee;
+    fee = `${+feeInNative <= 0.01 ? '<' : ''}${formatCurrency(feeInNative)}`;
+  } else {
+    // handle custom networks fee
```

Comment:
is this finished?

Reply:
yeah not sure how to do that yet, left a comment for later

Reply:
so we won't merge this yet?

Reply:
figured it out 🎉